llvm/lib/Target/AIE/AIEHazardRecognizerPRAS.cpp

```diff
@@ -24,7 +24,8 @@ using namespace llvm;
 
 AIEHazardRecognizerPRAS::AIEHazardRecognizerPRAS(const AIEBaseInstrInfo *TII,
                                                  const InstrItineraryData *II)
-    : AIEHazardRecognizer(TII, II, /*IsPreRA=*/false),
+    : AIEHazardRecognizer(TII, II, new AIEAlternateDescriptors(),
```

Comment:
Will that ever be freed?

Reply:
Will free in the destructor of  `AIEHazardRecognizerPRAS`